problem_builder/data_export.py

```diff
@@ -131,13 +156,34 @@ def _delete_export(self):
         self.active_export_task_id = ''
 
     @XBlock.json_handler
-    def start_export(self, request, suffix=''):
+    def start_export(self, data, suffix=''):
         """ Start a new asynchronous export """
+        block_types = data.get('block_types', None)
+        username = data.get('username', None)
+        root_block_id = data.get('root_block_id', None)
+        if not root_block_id:
+            root_block_id = unicode(self.scope_ids.usage_id)
+            get_root = True
+        else:
+            from xmodule.modulestore.django import modulestore
+            block = modulestore().get_items(self.runtime.course_id, qualifiers={'name': root_block_id}, depth=0)[0]
```

Comment:
I think it would be best to push this logic into `export_data_task`. XBlocks are not supposed to know about the modulestore, and all of the other modulestore logic is currently in that task.